### DISCUSSION TITLE
Way to retrieve current replica/node sockets inside specific room

### DISCUSSION THREAD

**Discussion by ghost** [socketio/socket.io#4792] [Q&A] [answered]
Hi, I have to retrieve all the sockets related to one specific node/replica (therefore BYPASSING the adapter) belonging to a specific room. I have seen that there is the method `fetchSockets()` but it works around all the nodes, is there any clean way to retrieve the only sockets belonging to a room that the current node is responsible of?
The only alternative I've thought of is looping through the `io.of(ns).sockets` and checking for each socket the registered rooms inside `socket.rooms`. Is this a valid workaround? Or there is a dedicated API for this purpose?

**Comment by darrachequesne** [comment, marked as answer]
Hi! You can use the `local` flag:

```js
const sockets = await io.local.in("the-room").fetchSockets();
```

Reference: https://socket.io/docs/v4/server-api/#serverfetchsockets

**Reply by ghost** [reply]
Thank you, perfect!
